### DISCUSSION TITLE
Custom image component for image_viewer_page

### DISCUSSION THREAD

**Discussion by matthinc** [immich-app/immich#226] [General]
Hi @alextran1502 ,

I'm currently thinking about improving the image_viewer_page. I played around a bit on my local computer. Adding the PhotoView back to the image_viewer_page seems to work fine and allows for panning and zooming which I think is a necessity.

Currently this page is built around the CachedNetworkImage which requires some "hacks" to allow for the transition between the preview- and full scale version of the picture. I think a custom built image-loader component would give us much more flexibility here and could also simplify the development of upcoming features.

I would be willing to develop such a component. As that would be quite a large task, I first would like your feedback on the idea. Do you like the idea in general? I also want to make sure that you don't work on something like that yourself to avoid duplicate work.

I'm looking forward to your response.
Matthias

**Comment by alextran1502** [comment]
Hi @matthinc 

Thank you for this, it was one of the items I want to do before the official release as well. The last time I check, the PhotoViewer library had an incompatibility issue with Flutter, 2.10 I believe. They might have fixed it with the latest Flutter 3. 

Please feel free to work on it and open a PR, I will put in feedback and thoughts as we go along :)
